CondFormats/CTPPSReadoutObjects/interface/LHCOpticalFunctionsSet.h

```diff
@@ -0,0 +1,63 @@
+#ifndef CondFormats_CTPPSReadoutObjects_LHCOpticalFunctionsSet_h
+#define CondFormats_CTPPSReadoutObjects_LHCOpticalFunctionsSet_h
+
+// Original Author:  Jan Kašpar
+
+#include "CondFormats/Serialization/interface/Serializable.h"
+
+#include <vector>
+#include <string>
+#include <memory>
+
+#include "TSpline.h"
+
+/// Set of optical functions corresponding to one scoring plane along LHC.
+class LHCOpticalFunctionsSet
+{
+  public:
+    /// indices for m_fcn_values and m_splines data members
+    enum { evx, eLx, e14, exd, evpx, eLpx, e24, expd, e32, evy, eLy, eyd, e42, evpy, eLpy, eypd };
+
+    LHCOpticalFunctionsSet() = default;
+    /// fills m_*_values fields from a ROOT file
+    LHCOpticalFunctionsSet(const std::string &fileName, const std::string &directoryName, double z);
+    ~LHCOpticalFunctionsSet() = default;
+
+    /// returns the position of the scoring plane (LHC/TOTEM convention)
+    double getScoringPlaneZ() const { return m_z; }
+    const std::vector<double>& getXiValues() const { return m_xi_values; }
+    const std::array<std::vector<double>, 16>& getFcnValues() const { return m_fcn_values; }
+    const std::array<std::shared_ptr<const TSpline3>, 16>& getSplines() const { return m_splines; }
```

Comment:
this is the first instance of using TSpline\* in CMSSW.
Do we have a simpler alternative?

Reply:
The only alternative I can think of is the GSL interpolation. However, I am not sure that it qualifies to be simpler.

Reply:
is there anything in Eigen?

Reply:
There is something, but the documentation is broken: http://eigen.tuxfamily.org/dox/group__Splines__Module.html
I will re-check later...